src/platforms/web/compiler/directives/html.js

```diff
@@ -2,7 +2,14 @@
 
 import { addProp } from 'compiler/helpers'
 
-export default function html (el: ASTElement, dir: ASTDirective) {
+export default function html (el: ASTElement, dir: ASTDirective, warn: Function) {
+  if (el.children.length) {
```

Comment:
Add `process.env.NODE_ENV !== 'production'` to ensure that this will only warn in non production enviroments.